public/src/client/category.js

```diff
@@ -125,26 +125,43 @@ define('forum/category', [
 
 		searchEl.on('keyup', function () {
 			const searchTerm = searchEl.val().toLowerCase();
+
 			topicEls.each(function () {
 				const topicEl = $(this);
 				const titleEl = topicEl.find('[component="topic/header"] a');
 				const title = titleEl.text().toLowerCase();
-				const isMatch = title.indexOf(searchTerm) !== -1;
-				console.log(`Title: ${title}\n ${isMatch ? 'Hidden' : 'Visible'}\n`);
-
-				topicEl.toggleClass('hidden', !isMatch);
+				const titleMatch = title.indexOf(searchTerm) !== -1;
+				const contentEl = topicEl.find('[component="topic/hidden-content"]');
+				const content = contentEl.text().toLowerCase();
+				const contentIdx = content.indexOf(searchTerm)
+				const hasMatch = (titleMatch || (contentIdx !== -1))
+				const searchContentEl = topicEl.find('[component="topic/search-content"]');
+
+				topicEl.toggleClass('hidden', !hasMatch);
+
+				// Use topic/search-content placeholder to display the 100 character preview around the search term. 
+				// Only do this if the search is more than 2 characters
+				if (contentIdx !== -1 && searchTerm.length >= 2) {
+					const start = Math.max(0, contentIdx - 50);
+					const end = Math.min(content.length, contentIdx + searchTerm.length + 50);
+					const snippet = content.substring(start, end);
+
+					const highlightedSnippet = snippet.replace(new RegExp(`(${searchTerm})`, 'gi'), '<strong class="highlight">$1</strong>');
```

Comment:
Code looks good; maybe include more comments like context of how highilghtedSnippet works

Reply:
Added some comments, thanks!